sos/plugins/libvirt.py

```diff
@@ -61,6 +64,12 @@ def setup(self):
 
         self.add_cmd_output("ls -lR /var/lib/libvirt/qemu")
 
+        # get details of processes of KVM hosts
+        for pidfile in glob.glob("/var/run/libvirt/*/*.pid"):
+            pid = open(pidfile).read().splitlines()[0]
+            for pf in ["environ", "cgroups", "maps", "numa_maps", "limits"]:
+                self.add_copy_spec("/proc/%s/%s" % (pid, pf))
```

Comment:
I'm a bit uncomfortable with this, but I'll allow it for now. I would really like to see a better solution before 3.4 is released, though.

My problem with this is it "crosses the streams"; generic process data is supposed to be collected by the `process` plugin and I try to avoid adding extra random bits in other plugins. That said, there are particular reasons why this data is relevant for users of qemu-kvm so I do understand the need.

Attempts to add generic `/proc` capture have so far floundered on performance problems on large hosts - if we can find a way to address that then I think we can remove this from the `libvirt` plugin.